src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -34,7 +34,7 @@ internal partial class Binder
             // Create a map from type parameter name to ordinal.
             // No need to report duplicate names since duplicates
             // are reported when the type parameters are bound.
-            var names = new Dictionary<string, int>(n);
+            var names = new Dictionary<string, int>(n, StringOrdinalComparer.Instance);
```

Comment:
@AlekseyTs That is correct, but I was under the impression we had a policy implemented last year that all comparisons must use an explicit comparer or culture, to prevent different defaults from floating through. Similar to our policy on `ConfigureAwait`.